[DOCS] Cross-reference API/deep dives/tutorials (#1229)
pytorch/torchtune: docs/source/deep_dives/checkpointer.rst

which supports the source format, without any code changes OR conversion scripts. This is one of the
ways in which torchtune interoperates with the surrounding ecosystem.

.. note::

  To be state-dict "invariant" in this way, the ``load_checkpoint`` and ``save_checkpoint`` methods of each checkpointer

>   make use of weight convertors which correctly map weights between checkpoint formats. For example, when loading weights

Reviewer: ```suggestion
  make use of weight converters which correctly map weights between checkpoint formats. For example, when loading weights
```

>   from HuggingFace, we apply a permutation to certain weights on load and save to ensure checkpoints behave exactly the same.

Reviewer: ```suggestion
  from Hugging Face, we apply a permutation to certain weights on load and save to ensure checkpoints behave exactly the same.
```
is permutation the right word? maybe "remap certain weights"?

>   To further illustrate this, the llama family of models uses a

Reviewer: ```suggestion
  To further illustrate this, the Llama family of models uses a
```

>   `generic weight converter function <https://github.com/pytorch/torchtune/blob/898670f0eb58f956b5228e5a55ccac4ea0efaff8/torchtune/models/convert_weights.py#L113>`_

Author: Yeah, ideally we expose this correctly through the docs. However, we don't generate docs for anything inside `convert_weights`. As mentioned [above](https://github.com/pytorch/torchtune/pull/1229#discussion_r1693148724), exposing `hf_to_tune` etc. in the `modules` api ref looks really weird, and, as Joe mentioned, we want to contextualise these functions to only be used inside of checkpointers - we don't want users thinking they have to convert weights using these functions.

I think I'd like to see another issue addressing where our model conversion logic lives so our api doctree looks sane.

>   whilst some other models like Phi have their own `conversion functions <https://github.com/pytorch/torchtune/blob/main/torchtune/models/phi3/_convert_weights.py>`_

Reviewer: ```suggestion
  whilst some other models like Phi3 have their own `conversion functions <https://github.com/pytorch/torchtune/blob/main/torchtune/models/phi3/_convert_weights.py>`_
```
